### DISCUSSION TITLE
Not sure which datalogger to use

### DISCUSSION THREAD

**Discussion by WimDeceuninck** [alienatedsec/solis-ha-modbus-cloud#40] [Q&A] [answered]
Hello,

I have a Solis RHI-3.6K-48ES-5G inverter and 2 FOX 5KW batteries (together 10KW).

With Home Assistant and an EPEX spot data sensor I control the charger for my EV (dynamic energy contract).

With the Solis app I can manually operate my inverter and enter charge and discharge times in function of the EPEX spot data price.

Now I 'd like to automate these charge and discharge times with HA SolaX modbus integration, but I do not understand the overview of the functions and dependencies. 

<img width="572" alt="image" src="https://github.com/user-attachments/assets/79feb883-c84c-4f46-bed8-d06c472248ac">

On the one hand the overview states that Home Assistant and Solis Cloud can work simultaneous with the compatible data loggers (yes 5* if combined with any compatible datalogger = DLS-W, DLS-L, S2-WL-ST and S3-WIFI-ST), but on the other hand, the same table states that both functions do not work simutaneous with the S3-WIFI-ST?

Which is very confusing to me, and I don't know if any of my dataloggers fits to achieve my goal.

I have 3 dataloggers here - one S3-WIFI-ST, one DLS version 1.2 (LAN) and one DLS version 2.2 (WIFI) - and I do not know if one of them is suitable to have both Home Assistant and Solis Cloud to work simultaneously. 

Except for the S3-WIFI-ST, the other type descriptions on the dataloggers I have do not match the types in the table.
Is DLS-L*1 my DLS version 1.2 (LAN) ?
And is DLS-W my DLS version 2.2 (WIFI) ?
Is my interpretation correct?

Can someone help me on my way please?

Regards,
Wim

**Comment by alienatedsec** [comment, marked as answer]
> I have 3 dataloggers here - one S3-WIFI-ST, one DLS version 1.2 (LAN) and one DLS version 2.2 (WIFI) - and I do not know if one of them is suitable to have both Home Assistant and Solis Cloud to work simultaneously.

Hi @WimDeceuninck 

Only `DLS-W` and `DLS-L` variants could work with HA and Solis Cloud simultaneously, and only if you have the right revision model. Not sure about the versions; however, it's easier to identify if you share their pictures. Think about using one of these dataloggers and nothing else.

However.....

What's good about this project is that you can use any datalogger just for Solis Cloud, (no matter which model) and combine it with Waveshares that work with Home Assistant; hence, why it is described this way.

**Reply by WimDeceuninck** [reply]
Hi Gordon,

Thank you for your quick reply and clarification.
Here are some pictures.
Currently I am using the DLS-W.
Initially the inverter was installed with the DLS-L which was replaced with the S3-WIFI-ST by the installer to allow them to upgrade the firmware of the inverter (if I remember well).
Last year I purchased another DLS-W hoping this one would allow to use the Solarman integration, but I did not get it work.

Happy to hear from you!

Regards, 
Wim

![20241101_085518](https://github.com/user-attachments/assets/67540297-8e02-48c0-8c8a-32b1e3033e07)
![20241101_084850](https://github.com/user-attachments/assets/b36e0871-34f7-4a21-b25d-6962d37fc14f)
![20241101_081608](https://github.com/user-attachments/assets/7560254b-b32f-4342-a956-d26a40e2037c)

**Reply by alienatedsec** [reply]
@WimDeceuninck nice collection ;)

So the `DLS-W` looks like a newer model that is known to be limited and no longer expose the SolarMan compatible "protocol" to work with HA. 

Furthermore, the `DLS-L` has an older serial number, which is known to be incompatible with Modbus TCP.

The `S3-WIFI-ST` is cloud only. 

The only other options I can recommend are to get another datalogger and hope to receive the correct one or following instructions in this project.

**Reply by WimDeceuninck** [reply]
Hi @alienatedsec, 
Yes, my collection is big enough, 😊
I'm not going to buy any new ones.
The chance that I will find a good one is almost non-existent.
Your method is challenging for me, -I'm just an amateur technician-; but since I want to use the inverter for another 5 years, I'm going to take the coming winter period and see if I can get the inverter control working here.
I will adhere to your installation procedure, that should work like a charm here too.
But I could probably use some advice or help.
I'll let you know.
Kind regards,
Wim
